pylint/config/_breaking_changes.py

```diff
@@ -0,0 +1,95 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/pylint-dev/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/pylint-dev/pylint/blob/main/CONTRIBUTORS.txt
+
+"""List the breaking changes in configuration files and their solutions."""
+
+from __future__ import annotations
+
+import enum
+from typing import NamedTuple
+
+
+class BreakingChange(enum.Enum):
+    MESSAGE_MADE_DISABLED_BY_DEFAULT = "{symbol} ({msgid}) was disabled by default"
+    MESSAGE_MADE_ENABLED_BY_DEFAULT = "{symbol} ({msgid}) was enabled by default"
+    MESSAGE_MOVED_TO_EXTENSION = "{symbol} ({msgid}) was moved to {extension}"
+    EXTENSION_REMOVED = "{extension} was removed"
+    # This kind of upgrade is non-breaking but if we want to automatically upgrade it,
+    # then we should use the message store and old_names values instead of duplicating
+    # MESSAGE_RENAMED= "{symbol} ({msgid}) was renamed"
+
+
+class Condition(enum.Enum):
+    MESSAGE_IS_ENABLED = "{symbol} ({msgid}) is enabled"
+    MESSAGE_IS_NOT_ENABLED = "{symbol} ({msgid}) is not enabled"
+    MESSAGE_IS_DISABLED = "{symbol} ({msgid}) is disabled"
+    MESSAGE_IS_NOT_DISABLED = "{symbol} ({msgid}) is not disabled"
+    EXTENSION_IS_LOADED = "{extension} is loaded"
+    EXTENSION_IS_NOT_LOADED = "{extension} is not loaded"
+
+
+class Information(NamedTuple):
+    symbol: str
+    msgid: str
+    extension: str | None
+
+
+class Solution(enum.Enum):
+    ADD_EXTENSION = "Add {extension} in 'load-plugins' option"
+    REMOVE_EXTENSION = "Remove {extension} from the 'load-plugins' option"
+    ENABLE_MESSAGE_EXPLICITLY = (
+        "{symbol} ({msgid}) should be added in the 'enable' option"
+    )
+    ENABLE_MESSAGE_IMPLICITLY = (
+        "{symbol} ({msgid}) should be removed from the 'disable' option"
+    )
+    DISABLE_MESSAGE_EXPLICITLY = (
+        "{symbol} ({msgid}) should be added in the 'disable' option"
+    )
+    DISABLE_MESSAGE_IMPLICITLY = (
+        "{symbol} ({msgid}) should be removed from the 'enable' option"
+    )
+
+
+ConditionsToBeAffected = list[Condition]
+# A solution to a breaking change might imply multiple actions
+MultipleActionSolution = list[Solution]
+# Sometimes there's multiple solutions and the user need to choose
+Solutions = list[MultipleActionSolution]
+BreakingChangeWithSolution = tuple[
+    BreakingChange, Information, ConditionsToBeAffected, Solutions
+]
+
+NO_SELF_USE = Information(
+    msgid="R6301", symbol="no-self-use", extension="pylint.extensions.no_self_use"
+)
+COMPARE_TO_ZERO = Information(
+    msgid="C2001", symbol="compare-to-zero", extension="pylint.extensions.comparetozero"
```

Comment:
What happens when the message ids change? (as we do somewhat frequently)

Reply:
Hmm... We're going to have to check with the message store (we also need the message store if we want to upgrade the message names automatically later). We have all the msgid/symbol correspondence in the message store so there's a little duplication. But not the "old extensions" infos. I wonder if we should just use ``msgid_or_symbol`` and add the ``old_extensions`` info in the message dict from the checker...

```suggestion
    msgid_or_symbol="C2001", extension="pylint.extensions.comparetozero"
```
```suggestion
    msgid_or_symbol=""compare-to-zero"
    
    #In the implicit_booleaness_checker .MSGS 
    ...
    old_extensions = [""pylint.extensions.comparetozero"]
    ...
```